Added missing sentiment classifier, fixes #1

haotian1028 opened the pull request.

Commit:
Added missing sentiment classifier, fixes #1

connorn-dev committed:
Restructured Notebook, added API folder

- Added API folder
- Train and Test notebook for each part of the process, to break it up

connorn-dev committed:
Changed Imports

- Cleaned up imports

carlosparadis commented on api/train.py:
Can you combine this to test.py and rename to model.py?

carlosparadis commented on notebooks/Software_Development_Sentiment_Classification.ipynb:
```diff
@@ -0,0 +1,239 @@
+        "id": "RAXtSnSK4LPr"
+      },
+      "source": [
+        "# Tokenlized\n",
```
Typo?

carlosparadis commented on notebooks/Software_Development_Sentiment_Classification.ipynb:
rename file to tokenize_statistics.ipynb

carlosparadis replied:
Note to self: This notebook does not tokenize anything currently, it just adds a column with 1s associated to the tokenizer. Not sure why, will figure out in the future.

carlosparadis reviewed (changes requested):
Thank you for moving the code around to the api folder. Only a few minor changes. When you looked at the original file, were there pip commands somewhere? 

I am not seeing anything that install the dependencies. Could you create an env.yml file here without using the export? (see my remark on env.yml file on the process mining project)

carlosparadis commented on api/filter.py:
Note to self: This filter.py functions are not used in the entire original notebook, and therefore it is not used in these 3 refactored notebooks.

carlosparadis commented on api/tokenizer.py:
Note to self: This tokenizer.py functions are not used in the entire original notebook, and therefore it is not used in these 3 refactored notebooks.

carlosparadis reviewed (changes requested):
We confirmed that the only files that will trully be needed on this notebook are:

- so-dataset.csv
- gh-dataset.csv
- crossplatform_sf_dataset.csv

and also the one column ones:


- so-dataset_tokenized.csv
- gh-dataset_tokenized.csv
- crossplatform_sf_dataset_tokenized.csv

carlosparadis commented:
So just to confirm I just need your pips if possible as a env.yml and to merge the api files of train.py and test.py to model.py

connorn-dev committed:
Added .env combined test.py and train.py

- train.py and test.py now exist in model.py
- env added for required packages
- Minor typo changes

Signed-off-by: Connor Narowetz <cnarowetz@gmail.com>

connorn-dev committed:
Added __init__.py to create pdoc and added model.py docs

- __init__.py added
- docs for model.py added

Signed-off-by: Connor Narowetz <cnarowetz@gmail.com>

connorn-dev commented:
### Pdocs Attached

[docs.zip](https://github.com/user-attachments/files/20117341/docs.zip)

carlosparadis commented:
@connorn-dev thank you for remembering this!

geraldmjhuff committed:
Modified api/model.py, added exec/train_or_predict.py

In api/model.py, added functions:
- train_model()
- predict_model()
Haotians original train_model() and test_model() commented out.

Added exec/train_or_predict.py:
- Python script to be syscalled from Kaiaulu
- Calls train_model() and predict_model()

carlosparadis committed:
Update env.yml to force Python and transformer ver

Non specified versions of python and transformers can
default transformers to several earlier versions, leading
to functions not being recognized.

This combination of python and transformers version works
as a standalone and called from kaiaulu system calls
via pysenti train / predict in conjunction with conda run
commands.

Signed-off-by: Carlos Paradis <carlosviansi@gmail.com>
